project_overview.md

```diff
@@ -0,0 +1,64 @@
+
+## CageConfigurator
+
+* __CageConfigurator__: Checks if it is running inside of a secure desktop and only starts the GUI component `CageConfiguratorForm` if the check was successful. Currently this check is done by comparing the access control list of the current desktop with the ACL set when the secure desktop is created, but it is planned to change this to getting the verification from the `CageService` using the unique identifier assigned to each new instance of the secure desktop.
+* __CageConfiguratorForm__: Lets the user configure applications to be run inside Shark Cage. The following options are available:
+  * Application (mandatory): Path to the application to be started inside the Shark Cage.
+  * Option to restrict returning from secure desktop to the exit button on the `CageLabeler` (if not checked the desktop will also close once all applications on it are closed). This option exists because there can be compatibility issues with some programs (e.g. Internet Explorer) where checking for the running program is not possible and the secure desktop would close immediately after opening.
+  * Secondary application: Sometimes it is useful to have an auxiliary program running besides the main application. This option currently only supports `Keepass`.
+  * Picture token (mandatory) for identifying the Shark Cage instance when run (picture is also displayed inside the secure desktop by the `CageLabeler`). This can either be from the users storage or directly captured by a connectec webcam.
+  * Name of configuration (mandatory). This name is used for the name of the config file and displayed by the `CageChooser`.
+
+## CageManager
+
+* __base64__: De- and encodes plain text from / into base64.
+* __CageDesktop__: Responsible for creating and switching to a new desktop and reserving space for the `CageLabeler`. When the secure desktop is no longer needed, this class also switches back to the previous desktop.
+* __CageLabeler__: Displays information about the currently running configuration: picture token, application names and hashes, activation buttons for the applications (to bring them in focus or restart them), exit restriction information, exit button.
+* __CageManager__: Immediately after getting started the `CageManager` listens for a tcp message. If no message arrives it just exits without doing anything else. If there is an expected message it parses it and then validates (signature / hash) all binaries specified in the reveived configuration. After validation it creates the security attributes with the `SecuritySetup`, creates a new access token with the `tokenLib` and a new desktop with a unique name & the security attributes. After this, the `CageLabeler` is used to display information on this new desktop. Once all this preparation is done the actual processes get started on the secure desktop and the `CageManager` waits for the correct exit conditions before cleaning up and exiting itself.
```

Comment:
...it creates the security attributes... should be `SECURITY_ATTRIBUTES` or?

Reply:
I thought about the same thing but came to the decision that in this instance I'm talking more about security attributes in general than the explicit data structure. But I can change it if you want.

Reply:
It's good so!
Thought it is exactly the `SECURITY_ATTRIBUTES`, because it's not it fits the way it is.